witness.md

```diff
@@ -195,8 +195,10 @@ Next, recursively define the encoding for an Ethereum state tree node, with some
 
 <Account_Node(d<65)> := 0x00 pathnibbles:<Nibbles(64-d)> address:<Address> balance:<Bytes32> nonce:<Bytes32>
                         {account node for externally owned account with values (pathnibbles, address, balance, nonce)}
-                      | 0x01 pathnibbles:<Nibbles(64-d)> address:<Address> balance:<Bytes32> nonce:<Bytes32> bytecode:<Bytecode> storage:<Account_Storage_Tree_Node(0)>
-                        {account node for contract account with values (pathnibbles address balance nonce bytecode storage)}
+                      | 0x01 pathnibbles:<Nibbles(64-d)> address:<Address> balance:<Bytes32> nonce:<Bytes32> code:<Bytecode> storage:<Account_Storage_Tree_Node(0)>
+                        {account node for executed contract account with values (pathnibbles address balance nonce code storage)}
+                      | 0x02 pathnibbles:<Nibbles(64-d)> address:<Address> balance:<Bytes32> nonce:<Bytes32> codehash:<Bytes32> codesize:<U32>
```

Comment:
This rule should also include `storage:<Account_Storage_Tree_Node(0)>`, since the storage root hash is needed to compute the merkle hash of a contract account.

Reply:
Thanks, this is a typo while I was copying things around.

Reply:
Actually I've removed storage intentionally. If the account is only touched, there is no way to inspect its storage. (Unless EIPs like EXTSLOAD are included in Eth1.)

Reply:
I still think that the storage hash is needed to merkleize the account. Otherwise, we can't verify the contents of the account against the world state merkle root.

Reply:
Sure, maybe it wasn't intuitive enough for me to pick up that `Account_Storage_Tree_Node` actually could also refer to a roothash.

Reply:
Fixed.